### PR TITLE
Adding The unsaturated callback to the queue #868

### PR DESCRIPTION
**Pull request by ecasilla**
I needed this function to allow be to create a simple backoff notification for the producers of data on to a queue.

### CODE REVIEW

File: lib/async.js [hunk 1]
```diff
@@ -879,6 +882,8 @@
             concurrency: concurrency,
             payload: payload,
             saturated: noop,
+            unsaturated:noop,
+            buffer: concurrency / 4,
```
Comment (aearly) [inline lib/async.js]: What is the point of the buffer?

Comment (ecasilla) [inline lib/async.js]: the difference between the saturation point and a buffer point you want in order to say the queue is unsaturated ...

Here I'm saying once the queue has been saturated i want it to be at 25% - concurrency in order to say the queue is unsaturated..

Example 
concurrency = 10 //workers
saturated = 10 //items in queue
unsaturated = 7.5 //items in queue or less..

**Comment by aearly** [comment]
If you can rebase from master,  I'll merge this.

**Comment by lucaswxp** [comment]
+1

**Comment by ecasilla** [comment]
ok will do

**Comment by ecasilla** [comment]
@aearly updated the branch

**Comment by lucaswxp** [comment]
Will this be merged into master?

2016-01-19 15:14 GMT-03:00 Ernie Casilla <notifications@github.com>:

**Comment by ecasilla** [comment]
yes i assume it will is there another branch i should be making the PR into?

**Comment by aearly** [comment]
We're currently going through a modularization phase (#996) so it would be more useful if  you could merge it into that branch (`modularization`).  You'll see there are lots of changes, so you'll likely have to re-do some of the work.

Sorry, things changed a lot since my last comment.

**Comment by lucaswxp** [comment]
This was closed without the PR?

**Comment by aearly** [comment]
It was completed in another PR.

**Comment by lucaswxp** [comment]
This is already available on master or should I checkout another branch?

**Comment by aearly** [comment]
It's on master.  We're probably going to do a v2.0.0-rc.2 soon.

**Pull request closed without merging**